index.bs

```diff
@@ -4969,162 +4907,193 @@ The [=map/values=] of the [=context menu map=], which are
 
-        : {{OnClickData/editable}}
+        : {{MenuItemDetails/editable}}
         :: Whether the |element| selected is editable, such as a text input.
 
-        : {{OnClickData/frameId}}
-        :: The [=navigable/frameId=] of the [=navigable container=] from which the [=/contextmenu=] [=event=] is dispatched.
+        : {{MenuItemDetails/frameId}}
+        :: The [=navigable/frameId=] of the [=navigable container=] from which
+            the [=/contextmenu=] [=event=] is dispatched.
 
-        : {{OnClickData/frameUrl}}
-        :: The [=Document/URL=] of the [= navigable container =] from which the [=/contextmenu=] [=event=] is dispatched.
+        : {{MenuItemDetails/frameURL}}
+        :: The [=Document/URL=] of the [=navigable container=] from which the
+            [=/contextmenu=] [=event=] is dispatched.
 
-        : {{OnClickData/menuItemId}}
+        : {{MenuItemDetails/id}}
         :: |id|.
 
-        : {{OnClickData/pageUrl}}
+        : {{MenuItemDetails/pageURL}}
         :: The [=Document/URL=] of the [=embedded navigable=].
 
-    1. Update the following optional fields of |info|:
+    1. Update the following optional fields of |menuItem|:
 
         Note: Different user agents may set these fields differently.
 
-        : {{OnClickData/checked}}
+        : {{MenuItemDetails/checked}}
         :: Whether the [=context menu item=] is checked.
 
-        : {{OnClickData/linkUrl}}
-        :: If |element| is an instance of {{HTMLAnchorElement}}, |element|'s [=href=] [=attribute=].
+        : {{MenuItemDetails/linkURL}}
+        :: If |element| is an instance of {{HTMLAnchorElement}}, |element|'s
+            [=href=] [=attribute=].
 
-        : {{OnClickData/mediaType}}
+        : {{MenuItemDetails/mediaType}}
         :: One of "image", "video", or "audio", if any, based on |element|.
 
-        : {{OnClickData/parentMenuId}}
+        : {{MenuItemDetails/parentMenuId}}
         :: |properties|["{{ContextMenusProperties/parentId}}"].
 
-        : {{OnClickData/selectionText}}
+        : {{MenuItemDetails/selectionText}}
         :: The text for the context selection, if any.
 
-        : {{OnClickData/srcUrl}}
+        : {{MenuItemDetails/srcURL}}
         :: The "src" attribute of |element|, if it exists.
 
-        : {{OnClickData/wasChecked}}
+        : {{MenuItemDetails/wasChecked}}
         :: The state of a checkbox or radio item before it was clicked.
 
-1. Let |callbackList| be the [=ContextMenusEvent/event listener list=] of |event|.
-
-1. [=list/For each=] |callback| in |callbackList|:
-
-  1. [=Queue a global task=] on the [=DOM manipulation task source=] of |controlledFrame|'s [=relevant global object=] to invoke |callback| with |info|.
+1. Let |event| be a [=new=] {{ContextMenusClickEvent}} named
+    [=ContextMenus/click=].
 
-1. If the {{ContextMenusProperties/onclick}} field of |properties| is not null:
+1. Set |event|["{{ContextMenusClickEvent/menuItem}}"] to |menuItem|.
```

Comment:
The fields of `MenuItemDetails` are a mix of details about the menu item (like `id` and `parentMenuId`) and details about the thing that was clicked on. The latter seem like properties that should be on the event rather than the menu item interface.